### ISSUE TITLE
Trusted domains turning into allowed only after update to 1.0.7.0

### ISSUE THREAD

**Issue by ROpdebee** [andryou/scriptsafe#9]
I've noticed you've revamped the domain matching logic, great work. However, the migration from the previous version to the new domain matching logic didn't seem to go all too smooth, as all my previously trusted domains have now been turned into just "allowed" domains.

Now, this doesn't cause too much issues, but it does cause some:
- localhost was a previously trusted domain because I have a few apps with web interfaces running on them, now its pattern `*.localhost` doesn't allow any script to run on localhost (as it's not `www.localhost`)
- Whenever I visit a site that load scripts with URLs such as `subdomain-of-subdomain.subdomain.example.com` and I had the pattern `*.example.com`, previously I'd expect the script to be loaded, but since 1.0.7.0 it won't because of the new pattern matching system.

Of course, the above are rather rare issues, but in the panel these previously trusted domains also show up as just allowed.  For me, it's not a big issue as I've just purged my whitelist completely since the previous update because it was cluttered with old rules and now it's relatively empty, so it's reasonably easy to just update all patterns, but for other users this might be an issue.

For the majority of the users, it's already too late by now to fix this, but I advise to set up some kind of migration from old patterns to new ones if you update the pattern matching system again in the future.

Thanks for the efforts, been using this extension for ages now and I really like the new matching system 😄

**Comment by andryou** [comment]
Thanks for this! I had actually coded a one-time whitelist/blacklist update script that would run to convert all existing trusted entries (*.) to the new syntax (**.), but I realized I misset the check and it never fired.

I made a set of significant optimizations to the ScriptSafe UI panel component and am pushing another update in the next half-hour.

I will keep this issue open for you to review the new release when it's propagated! I will comment again here soon to let you know when I've pushed the update.

**Comment by ROpdebee** [comment]
Additionally, this may or may not be related: 
![screen shot 2016-05-31 at 02 51 54](https://cloud.githubusercontent.com/assets/15186467/15660307/d2b54652-26da-11e6-9bcf-fde8ec924dd4.png)
Shouldn't the icon be green in this case?

**Comment by andryou** [comment]
I couldn't reproduce the icon issue, but feel free to create another issue if you encounter it again.

I've just pushed v1.0.7.1, so you should be receiving it within a couple of hours! Feel free to close this when you receive it. Exact fix (very embarrassing for me) is here: https://github.com/andryou/scriptsafe/commit/c5fa25d43a074a50d1bdcbf94e2ad7c6220432c6#diff-37bff41a0205c5d9293fe828bf2631f3R800

**Comment by ROpdebee** [comment]
Yeah, it's fine now. My whitelist seems to be correct now. Thanks!
